contracts/drafts/dividend/DividendableERC20.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+
+/**
+ * @title DividendableERC20
+ * @notice Implements an ERC20 mintable token with a dividend distribution procedure
+ */
+contract DividendableERC20 is IERC20 {
+
+    using SafeMath for uint;
+
+    uint pointMultiplier = 10e18;
+    uint totalDividends;
+    uint totalDividendPoints;
+    mapping(address => uint) lastDividendPoints;
+
+    constructor() public {}
+
+    /**
+     * @dev Fallback function
+     * @notice Send ether to this contract in orther to disburse dividends
+     */
+    function () external payable {
+        totalDividends = totalDividends.add(msg.value);
```

Comment:
The contract receives ether, but distributes ERC20 tokens (which aren't minted anywhere).

I would suggest that the contract should receive ERC20 tokens (of itself), and we can leave to the user the question of how to mint them.

Reply:
I was wrong - from slack: "Think about it like this: I have 40% shares (i.e., tokens) in a venture (i.e. our contract), and you have the rest (60%). Now, the venture makes $10,000 in profit (10 ether) this year. When I claim my dividends (which I can do at any time, whenever the firm registers revenue) I expect to receive 4 ethers (and you, 6). Now, what's the point of distributing as dividends the shares of the same company? When any organization distributes dividends, it distributes the profit based on ANOTHER metric, i.e. the shares percentage"